lib/adapter.test.js

```diff
@@ -0,0 +1,66 @@
+const KnexAdapter = require('./adapter.js');
+const knex = require('knex');
+const mockKnex = require('mock-knex');
+const tracker = mockKnex.getTracker();
+
+function createMockPolicy(policyValues, roleValues) {
+  // Don't alter passed values
+  policyValues = [...policyValues];
+  roleValues = [...roleValues];
+  // Fill in missing values: v0 - v5
+  for (let index = 0; index < 6; index++) {
+    if (typeof policyValues[index] === 'undefined') {
+      policyValues[index] = null;
+    }
+
+    if (typeof roleValues[index] === 'undefined') {
+      roleValues[index] = null;
+    }
+  }
+
+  return {
+    model: new Map([
+      [
+        'p',
+        new Map([
+          [
+            'p',
+            {
+              policy: [policyValues]
+            }
+          ]
+        ])
+      ],
+      ['g', new Map([['g', { policy: [roleValues] }]])]
+    ])
+  };
+}
+
+describe('adapter.js', () => {
+  let knexAdapter;
+  let database;
```

Comment:
You won't use the database in the tests anyways, so let's move this inside the `beforeAll`.